### ISSUE TITLE
Missing wraper for promiscous mode.

### ISSUE THREAD

**Issue by Frostie314159** [esp-rs/esp-idf-svc#246]
The enablement of promiscous mode isn't wrapped in a safe function.

**Comment by Vollbrecht** [comment]
do you want to create an PR for it ?

**Comment by Frostie314159** [comment]
Of course. Btw, ever noticed that the promisc mode described in the docs is actually monitor mode?

**Comment by Vollbrecht** [comment]
oh that sounds interesting. i never played with both before on esp, still on my todo list. does the esp driver can sniff on other packets while being an ap itself and send data to clients or does it come with some limitations?

**Comment by owenthewizard** [comment]
Promiscuous mode for Eth would be nice too.
